app/tests/spec/lib/experiments/grouping-rules/connect-another-device-on-signin.js

```diff
@@ -8,36 +8,71 @@ define(function (require, exports, module) {
   const { assert } = require('chai');
   const Account = require('models/account');
   const Experiment = require('lib/experiments/grouping-rules/connect-another-device-on-signin');
+  const sinon = require('sinon');
 
   describe('lib/experiments/grouping-rules/connect-another-device-on-signin', () => {
     let account;
     let experiment;
+    let validSubject;
 
-
-    before(() => {
-      account = new Account();
+    beforeEach(() => {
+      account = new Account({ email: 'testuser@testuser.com' });
       experiment = new Experiment();
+      validSubject = {
+        account,
+        env: 'production',
+        uniqueUserId: 'id'
+      };
     });
 
     describe('choose', () => {
-      it('returns false if no subject, or subject.account, or subject.account.email', () => {
+      it('returns false if invalid subject', () => {
         assert.isFalse(experiment.choose());
         assert.isFalse(experiment.choose({}));
-        assert.isFalse(experiment.choose({ account: new Account() }));
-      });
 
-      it('returns true for test email addresses', () => {
-        account.set('email', 'testuser@testuser.com');
-        assert.isFalse(experiment.choose({ account }));
+        assert.isFalse(experiment.choose(Object.assign(validSubject, { env: undefined })));
+        assert.isFalse(experiment.choose(Object.assign(validSubject, { uniqueUserId: undefined })));
```

Comment:
Is this `Object.assign` doing what you want it to? I'd have thought you want to test each invalid property in isolation, but by modifying `validSubject` in place like this it seems like you're polluting subsequent assertions with the the target of the earlier ones. Or am I missing something?

Reply:
Oh dear lord, you are right!